contracts/REXUniswapV3Market.sol

```diff
@@ -223,62 +195,55 @@ contract REXUniswapV3Market is
     /// @param _uniswapRouter is the Uniswap V3 Router
     /// @param _uniswapFactory is the Uniswap V3 Factory
     /// @param _uniswapPath is the Uniswap V3 path
-    /// @param _poolFee is the Uniswap V3 pool fee
+    /// @param _poolFees is the Uniswap V3 pool fees
+    /// @param _gasPoolFee is the Uniswap V3 pool fee for gas reimbursements
     function initializeUniswap(
         ISwapRouter02 _uniswapRouter,
         IUniswapV3Factory _uniswapFactory,
         address[] memory _uniswapPath,
-        uint24 _poolFee
+        uint24[] memory _poolFees,
+        uint24 _gasPoolFee
     ) external onlyOwner {
+
+        require(address(router) == address(0), "IU"); // Blocks if already initialized
+
+        // Set contract variables
         router = _uniswapRouter;
-        poolFee = _poolFee;
+        poolFees = _poolFees;
         uniswapPath = _uniswapPath;
+        gasPoolFee = _gasPoolFee;
 
         // Get the pool from the Uniswap V3 Factory
         IUniswapV3Factory factory = IUniswapV3Factory(_uniswapFactory);
 
         // Require that the pool for input/output swaps exists
-        require(
-            factory.getPool(
-                address(underlyingInputToken),
-                address(underlyingOutputToken),
-                poolFee
-            ) != address(0),
-            "PDNE1"
-        );
+        for(uint i = 0; i < _uniswapPath.length - 1; i++)
+            require(
+                factory.getPool(
+                    address(_uniswapPath[i]),
+                    address(_uniswapPath[i+1]),
+                    poolFees[i]
+                ) != address(0),
+                "PDNE"
+            );
 
         // Require that the pool for gas reimbursements exists
         if (address(underlyingInputToken) != address(wmatic)) {
             require(
                 factory.getPool(
                     address(wmatic),
                     address(underlyingInputToken),
-                    poolFee
+                    _gasPoolFee
```

Comment:
maybe make it `gelatoGasPoolFees`, might get confusing for someone looking at uniswap `poolFees`